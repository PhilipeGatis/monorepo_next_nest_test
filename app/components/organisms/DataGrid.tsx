import * as React from 'react'
import Box from '@mui/material/Box'
import Typography from '@mui/material/Typography'
import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell from '@mui/material/TableCell'
import TableContainer from '@mui/material/TableContainer'
import TableHead from '@mui/material/TableHead'
import TableRow from '@mui/material/TableRow'
import Paper from '@mui/material/Paper'
import { ProductListGroupedByDescriptionResponseDto } from '@hubla/api/dist/models/product/dto/ProductListGroupedByDescriptionResponse.dto'
import CollapsibleRow from '../molecules/CollapsibleRow'
import UploadButton from '../molecules/UploadButton'

const DataGrid: React.FC<{ data: ProductListGroupedByDescriptionResponseDto[] }> = ({ data }) => {
  return (
    <TableContainer component={Paper}>
      <Table aria-label="collapsible table">
        <TableHead>
          <TableRow>
            <TableCell />
            <TableCell>Produto</TableCell>
            <TableCell align="right">total</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {!!data.length ? (
            data.map(row => <CollapsibleRow key={row.title} row={row} />)
          ) : (
            <TableRow sx={{ height: '400px' }}>
              <TableCell colSpan={3} align="center">
                <Box sx={{ mb: 2 }}>
                  <Typography variant="h6">Lista vazia</Typography>
                  <Typography variant="caption">Fa√ßa o upload de uma lista de produtos.</Typography>
                </Box>
                <div>
                  <UploadButton />
                </div>
              </TableCell>
            </TableRow>
          )}
        </TableBody>
      </Table>
    </TableContainer>
  )
}

export default DataGrid
