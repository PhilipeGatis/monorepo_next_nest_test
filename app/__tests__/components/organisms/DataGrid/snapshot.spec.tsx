import { render } from '@testing-library/react'
import DataGrid from '../../../../components/organisms/DataGrid'
import mockData from './mock'

describe('DataGrid', () => {
  it('renders empty', () => {
    const { container } = render(<DataGrid data={[]} />)
    expect(container).toMatchSnapshot()
  })

  it('renders a heading', () => {
    const { container } = render(<DataGrid data={mockData} />)
    expect(container).toMatchSnapshot()
  })
})
