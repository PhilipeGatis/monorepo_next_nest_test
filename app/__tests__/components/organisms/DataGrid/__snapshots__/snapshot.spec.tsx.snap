// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid renders a heading 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="collapsible table"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          />
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Produto
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            total
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              data-testid="expandButton"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </td>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            CURSO DE BEM-ESTAR
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            R$ 25.500,00
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colspan="6"
            style="padding-bottom: 0px; padding-top: 0px;"
          />
        </tr>
        <tr
          class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              data-testid="expandButton"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </td>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            DOMINANDO INVESTIMENTOS
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            R$ 100.000,00
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colspan="6"
            style="padding-bottom: 0px; padding-top: 0px;"
          />
        </tr>
        <tr
          class="MuiTableRow-root css-2blr9s-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              data-testid="expandButton"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </td>
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            DESENVOLVEDOR FULL STACK
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            R$ 310.000,00
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            colspan="6"
            style="padding-bottom: 0px; padding-top: 0px;"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DataGrid renders empty 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="collapsible table"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          />
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Produto
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
          >
            total
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-16wuvdb-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
            colspan="3"
          >
            <div
              class="MuiBox-root css-1qm1lh"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                Lista vazia
              </h6>
              <span
                class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              >
                Faça o upload de uma lista de produtos.
              </span>
            </div>
            <div>
              <input
                accept="text/plain"
                id="contained-button-file"
                style="display: none;"
                type="file"
              />
              <label
                for="contained-button-file"
              >
                <span
                  class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                  id=":r0:"
                  role="button"
                  tabindex="0"
                >
                  Upload
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </label>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
